src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java

```diff
@@ -44,11 +45,11 @@ public void addAll(Map<E, Integer> possibilities) {
      * Generates a random element from this list, considering the specified possibilities.
      *
      * @return A random element
-     * @throws IndexOutOfBoundsException If this list is empty
      */
-    public E nextElement() throws IndexOutOfBoundsException {
+    public Optional<E> nextElement() throws IndexOutOfBoundsException {
+        if (underlyingList.isEmpty()) return Optional.empty();
         int randomIndex = ThreadLocalRandom.current().nextInt(underlyingList.size());
-        return underlyingList.get(randomIndex);
+        return Optional.of(underlyingList.get(randomIndex));
```

Comment:
Needs to be Optional.ofNullable cuz Optionals are dum

Reply:
It can't be null tho?

Reply:
The maximum of the index is always `underlyingList.size() - 1` and if the list is empty there is always an empty optional returned

Reply:
then why is it an optional if Optional.isPresent is always gonna be true?

Reply:
It's not, check line 50

Reply:
nevermind im blind